interpreter/cling/test/DynamicLibraryManager/callable_lib_L_AB_abs.C

```diff
@@ -10,7 +10,7 @@
 // RUN: mkdir -p %t-dir/lib
 // RUN: %clang -shared -DCLING_EXPORT=%dllexport %S/call_lib_A.c -o%t-dir/rlib/libcall_lib_A%shlibext
 // RUN: %clang -shared -DCLING_EXPORT=%dllexport %S/call_lib_B.c -o%t-dir/rlib/libcall_lib_B%shlibext
-// RUN: %clang %fPIC -shared -DCLING_EXPORT=%dllexport %S/call_lib_L_AB.c -o%t-dir/lib/libcall_lib_L_AB%shlibext -L %t-dir/rlib -l%t-dir/rlib/libcall_lib_A.so -l%t-dir/rlib/libcall_lib_B.so
+// RUN: %clang %fPIC -shared -DCLING_EXPORT=%dllexport %S/call_lib_L_AB.c -o%t-dir/lib/libcall_lib_L_AB%shlibext %t-dir/rlib/libcall_lib_A%shlibext %t-dir/rlib/libcall_lib_B%shlibext
```

Comment:
Why we dropped `-L`?

@alexander-penev, is the relative library path here relevant for the test?

Reply:
The `-L` isn't needed because we're passing absolute paths for the libraries. The absolute paths for them is important as far as I can tell, for Cling to find the library.

Reply:
Do we understand why this does not or no longer work? (i.e. is it "temporary" or a feature that will never be implemented?)

Reply:
@vgvassilev This test require absolute paths.

@hahnjo Absolute paths are not required by Cling. Just this test checks if the system works in this case. If removing -L does not change the semantics of all supported platforms, then this key can be removed so as not to interfere with some versions of the linkers.

Reply:
> Do we understand why this does not or no longer work? (i.e. is it "temporary" or a feature that will never be implemented?)

As far as I could tell from my system, passing an absolute path to `-l` (lower-case `L`) doesn't work at all with my version of `ld` (from CentOS 8 Stream) and neither does it on one of our Macs that I picked at random. It's possible that it works with some other version, though...

> Absolute paths are not required by Cling. Just this test checks if the system works in this case. If removing -L does not change the semantics of all supported platforms, then this key can be removed so as not to interfere with some versions of the linkers.

I don't agree here: If I do `%clang %fPIC -shared -DCLING_EXPORT=%dllexport %S/call_lib_L_AB.c -o%t-dir/lib/libcall_lib_L_AB%shlibext -L %t-dir/rlib/ -lcall_lib_A -lcall_lib_B`, ie without the absolute paths and using `-L` to find the libraries, Cling complains:
```
cling::DynamicLibraryManager::loadLibrary(): libcall_lib_A.so: cannot open shared object file: No such file or directory
```
So the absolute paths *are* important. Regarding the removal of `-L`: If you pass in the absolute paths, there is no point in adding the same directory to the library search path.

Reply:
I clarified with Alexander over IM - the change is okay.

> > Absolute paths are not required by Cling. Just this test checks if the system works in this case. If removing -L does not change the semantics of all supported platforms, then this key can be removed so as not to interfere with some versions of the linkers.
> 
> I don't agree here: If I do `%clang %fPIC -shared -DCLING_EXPORT=%dllexport %S/call_lib_L_AB.c -o%t-dir/lib/libcall_lib_L_AB%shlibext -L %t-dir/rlib/ -lcall_lib_A -lcall_lib_B`, ie without the absolute paths and using `-L` to find the libraries, Cling complains:
> 
> ```
> cling::DynamicLibraryManager::loadLibrary(): libcall_lib_A.so: cannot open shared object file: No such file or directory
> ```
> 
> So the absolute paths _are_ important. Regarding the removal of `-L`: If you pass in the absolute paths, there is no point in adding the same directory to the library search path.


FWIW, this is exactly what the tests tests for. It logic checks the RPATH contents among other things and that's expected behavior.

Reply:
It doesn't build before the change so I'm not sure how I'm supposed to do this?!

Reply:
Ah apologies, I will try this locally - please go ahead and merge this change.